assets/js/phoenix.js

```diff
@@ -854,7 +854,9 @@ export class Socket {
     this.log("transport", "close", event)
     this.triggerChanError()
     clearInterval(this.heartbeatTimer)
-    this.reconnectTimer.scheduleTimeout()
+    if (!(event instanceof Event) || ((event instanceof Event) && event.code !== 1000)) {
```

Comment:
do we need the instanceof checks? Can we simply say if `event && event.code !== 1000`

Reply:
@chrismccord yeap, it should work, but I have to refactor the test since they are passing a string on `event`. this line [https://github.com/phoenixframework/phoenix/blob/master/assets/test/socket_test.js#L645](https://github.com/phoenixframework/phoenix/blob/master/assets/test/socket_test.js#L645)

I'm gonna do it as soon as I can.

thanks!